packages/analytics-plugin-google-global-site-tag/src/browser.ts

```diff
@@ -0,0 +1,528 @@
+
+  const trackingId = pluginConfig.trackingId;
+
+  return {
+    name: 'google-gtag-analytics',
+    config: {
+      ...defaultConfig,
+      ...pluginConfig
+    },
+    
+    // Load gtag.js and define gtag
+    // Set custom dimensions (ua properties) and parameters (ga4 properties)
+    initialize: (pluginApi: IPluginApiProps) => {
+      const { config, instance } = pluginApi;
+      if (!trackingId) throw new Error('No GA trackingId defined');
+
+      const gtagScriptSource = 'https://www.googletagmanager.com/gtag/js';
+      const scriptSrc = config.customScriptSrc || `${gtagScriptSource}?id=${trackingId}`;
+
+      if (!gtagLoaded(config.customScriptSrc || gtagScriptSource)) {
+        injectScript(scriptSrc);
+      }
+
+      if (!window.gtag) {
+        setUpWindowGtag();
+      }
+
+      /**
+       * Covert custom dimensions to:
+       * { dimension1: traitOne, dimension2: traitTwo }
+       */
+      const customDimensions = formatCustomDimensionsIntoCustomMap(config);
+
+      /**
+       * Convert `cookieConfig`'s keys from camel case to snake cases
+       */
+      let newCookieConfig: { [key: string]: unknown} = {};
+      if (config.cookieConfig !== undefined) {
+        Object.entries(config.cookieConfig).forEach(([key, value]) => {
+          newCookieConfig[camelToSnakeCase(key)] = value;
+        })
+      }
+
+      // Initialize tracker instance on page
+      if (!loadedInstances[instanceName]) {
+        let gtagConfig: { [key: string]: unknown } = {
+          cookie_domain: config.domain || 'auto',
+          send_page_view: config.sendPageView ? config.sendPageView : true,
+          allow_google_signals: config.allowGoogleSignals,
+          allow_ad_personalization_signals: config.allowAdPersonalizationSignals,
+          custom_map: customDimensions,
+          anonymize_ip: config.anonymizeIp,
+          ...newCookieConfig,
+        }
+
+        if (config.linker) {
+          gtagConfig.linker = config.linker;
+        }
+
+        /* set custom dimensions from user traits */
+        const user = instance.user() || {}
+        const traits = user.traits || {}
+
+        if (Object.keys(traits).length) {
+          window.gtag('set', traits);
+        }
+
+        window.gtag('config', trackingId, gtagConfig);
+
+        loadedInstances[instanceName] = true;
+      }
+    },
+
+    // Set parameter scope at user level with 'set' method
+    identify: (props: IProps) => {
+      const { payload, config } = props;
+      identifyVisitor(payload.userId, payload.traits, config)
+    },
+
+    // Set parameter scope at page level with 'config' method
+    page: ({ payload, config, instance }: IProps) => {
+      if (!window.gtag || !config.trackingId) return;
+
+      const { properties } = payload;
+      const { resetCustomDimensionsOnPage, customDimensions } = config;
+      const campaign = instance.getState('context.campaign');
```

Comment:
Don't know too much about what campaign is. I simply copy pasted it from `@analytics-plugin-google-analytics/src/browser.js`